client/app/services/project.service.js

```diff
@@ -224,20 +252,112 @@
             }
 
             // check for self-intersections
+            for (var j = 0; j < features.length; j++) {
```

Comment:
j and k counters would be better named something more readable:

taskCount, polyCount or whatever